.github/workflows/process.yml

```
@@ -0,0 +1,66 @@
+name: Process newly added JSON
+
+on:
+  pull_request:
+    types: [closed]
+    branches:
+      - main
+
+permissions:
+  contents: write
+
+concurrency:
+  group: process-json-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  process-json:
+    # Only run if the PR was merged
+    if: github.event.pull_request.merged == true
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          # Needed to push changes back to the repo
+          fetch-depth: 0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "22"
+          cache: "yarn"
+          cache-dependency-path: actions/yarn.lock
+
+      - name: Install Dependencies
+        run: yarn install --frozen-lockfile
+        working-directory: ./actions
+
+      - name: Find newly added JSON files
+        id: find-json
+        run: |
+          # Get the list of added JSON files in the records/new/ directory
+          ADDED_FILES=$(git diff ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} --diff-filter=A --name-only | grep '^records/new/.*\.json$')
```

If you run on the `push` event, you can use `git show HEAD`
```suggestion
          ADDED_FILES=$(git show HEAD --format= --diff-filter=A --name-only | grep '^records/new/.*\.json$')
```

---

Only if it's squashed, it can be rather confusing if one made a mistake and didn't squash.

---

We should probably enable the GitHub setting that enforces linear history regardless of that suggestion.